plat/kvm/efi.c

```diff
@@ -16,6 +17,9 @@ static enum uk_efi_alloc_type uk_efi_alloc_type = UK_EFI_ALLOCATE_ANY_PAGES;
 #else  /* !CONFIG_HAVE_PAGING */
 static enum uk_efi_alloc_type uk_efi_alloc_type = UK_EFI_ALLOCATE_MAX_ADDRESS;
 #endif /* !CONFIG_HAVE_PAGING */
+#if CONFIG_LIBUKDEBUG_PRINTD
```

Comment:
Let's move this higher up together with the  other `#include`s. Also, if possible, try to have them in an alphabetic order, i.e. `uk/libid.h` would go after `uk/arch/paging.h`

Reply:
I also moved `uk_efi_printf ` out of the ifdef to have a leaner ifdef.

Reply:
BTW, what is the correct practice, should function definitions be inside or outside of ifdefs?

Reply:
I guess it depends on the case right? :thinking: But I am not sure I understand your question. In this case it is fine to have it outside I would say (for now) as there is nobody calling it directly really and does not have any dependency. But now that you mention it I am starting to have second thoughts :sweat_smile:. Since that function is not really used it occupies unnecessary space in the image. But that is so minor... and printf is such a generically available function as well... maaan I am really not sure, really getting second thoughts now :rofl:. I hate big ifdefs too... oh well let's just put it back for now so that it is implied that printing in efi stub is depndent on PRINTD config? What do you think? I just think that it is best to keep the efi stub as tiny as possible, although that function is not of that much impact either IMO...

Reply:
I'm asking because sometimes I see function definitions that are related to a config but are not wrapped between ifdefs, and sometimes they are 🤣. So, I'm unsure what the correct practice here is. About the code side, I think the compiler will optimize unused functions out if there is no caller, but I'm not 100% sure.